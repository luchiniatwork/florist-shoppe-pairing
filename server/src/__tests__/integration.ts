import { createTestClient } from 'apollo-server-testing';
import gql from 'graphql-tag';
// import nock from 'nock';

import { constructTestServer } from './__utils';

const GET_FLOWERS = gql`
  query {
    flowers {
      name
    }
  }
`;

const GET_FAMILIES = gql`
  query {
    families {
      name
      flowers {
        name
      }
    }
  }
`;

describe('Queries', () => {
  let server;
  beforeAll(async () => {
    server = await constructTestServer();
  });

  it('fetches a list of flowers', async () => {
    const { query } = createTestClient(server);
    const res = await query({ query: GET_FLOWERS });
    expect(res).toMatchSnapshot();
  });

  it('fetches a list of families', async () => {
    const { query } = createTestClient(server);
    const res = await query({ query: GET_FAMILIES });
    expect(res).toMatchSnapshot();
  });
});
