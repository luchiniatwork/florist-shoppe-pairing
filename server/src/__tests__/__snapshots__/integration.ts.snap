// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Queries fetches a list of families 1`] = `
Object {
  "data": Object {
    "families": Array [
      Object {
        "flowers": Array [
          Object {
            "name": "Irises",
          },
          Object {
            "name": "Gladiolus",
          },
          Object {
            "name": "Coppertips",
          },
          Object {
            "name": "Freesia",
          },
        ],
        "name": "Iridaceae",
      },
      Object {
        "flowers": Array [
          Object {
            "name": "Bellflowers",
          },
          Object {
            "name": "Platycodon grandiflorus",
          },
        ],
        "name": "Campanulaceae",
      },
      Object {
        "flowers": Array [
          Object {
            "name": "Lily",
          },
          Object {
            "name": "Tulip",
          },
          Object {
            "name": "Fritillaries",
          },
          Object {
            "name": "Fawn lilies",
          },
        ],
        "name": "Liliaceae",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Queries fetches a list of flowers 1`] = `
Object {
  "data": Object {
    "flowers": Array [
      Object {
        "name": "Irises",
      },
      Object {
        "name": "Gladiolus",
      },
      Object {
        "name": "Coppertips",
      },
      Object {
        "name": "Freesia",
      },
      Object {
        "name": "Bellflowers",
      },
      Object {
        "name": "Platycodon grandiflorus",
      },
      Object {
        "name": "Lily",
      },
      Object {
        "name": "Tulip",
      },
      Object {
        "name": "Fritillaries",
      },
      Object {
        "name": "Fawn lilies",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
